### DISCUSSION TITLE
Issue with patches/crops of a larger image while loading them into 3D Slicer or MITK.

### DISCUSSION THREAD

**Discussion by lenlobo** [Project-MONAI/MONAI#6451] [Q&A]
I was trying to patch or crop some larger volumes into slices of 100 voxels, at the moment I have achieved to do so, but I have noticed that some of the outputs are I can load and observe in a Slicer Docker Image, and MITK as well. But some I cannot, it throw's an orthogonal information missing error. Example, if I have 15 slices of a volume, the first 8-9 slices will load easily into Slicer, but the last 5-6 won't.
```
trans = Compose([
    LoadImage(image_only=True),
    ToTensor(),
    GridPatch(patch_size=100,overlap=0.2),
    SplitDim(dim=0)
])
img = trans(data)
orig_affine = img[0][0].affine
size=512//len(img[0])
for i in range(len(img[0])):
    newaffine = orig_affine
    newaffine[1,3] = i*20*size*orig_affine[1,1] #multiplying by 20 to give some space between the crops when I display them together

final_img= nib.Nifti1Image(np.asarray(torch.squeeze(img[0][i])),newaffine)
nib.save(final_img,'path/test{}.nii.gz'.format(i))

```
In the code, I change the affine at [1,3] so I can set my own affine for each crop with a slightly different reference location for each crop. I was looking for a method to automate the assigning of affine to the crops, but couldn't find anything. If I don't set an affine, all the crops have the same affine as the original image and they just overlap on top of each other.

As I mentioned, the issue here is, I can load and observe some of the crops, but some don't load. What could the reason be? Suggestions are welcomed as well to a better approach.

**Comment by wyli** [comment]
thanks for the question, instead of assigning affine[1,3], you can consider making a proper shifting matrix and then do `orig_affine @ shift` like this example:

https://github.com/Project-MONAI/MONAI/blob/c2a9a31beb22e1d3321016be7e68e0875cf2a8ad/monai/transforms/utility/array.py#L387-L390
